ksqldb-rest-app/src/main/java/io/confluent/ksql/api/server/SniHandler.java

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright 2022 Confluent Inc.
+ *
+ * Licensed under the Confluent Community License (the "License"); you may not use
+ * this file except in compliance with the License.  You may obtain a copy of the
+ * License at
+ *
+ * http://www.confluent.io/confluent-community-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OF ANY KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package io.confluent.ksql.api.server;
+
+import static io.netty.handler.codec.http.HttpResponseStatus.MISDIRECTED_REQUEST;
+
+import io.confluent.ksql.rest.Errors;
+import io.vertx.core.Handler;
+import io.vertx.ext.web.RoutingContext;
+
+public class SniHandler implements Handler<RoutingContext> {
+
+  public SniHandler() {
+  }
+
+  @Override
+  public void handle(final RoutingContext routingContext) {
+    if (routingContext.request().isSSL()) {
+      final String indicatedServerName = routingContext.request().connection()
+          .indicatedServerName();
+      final String requestHost = routingContext.request().host();
+      if (indicatedServerName != null && requestHost != null) {
+        // sometimes the port is present in the host header, remove it
+        final String requestHostNoPort = requestHost.replaceFirst(":\\d+", "");
+        if (!requestHostNoPort.equals(indicatedServerName)) {
```

Comment:
How confident are we that these two are expected to be the same? I wonder if there are other "got'chas" similar to the request host sometimes containing the port. I haven't drilled into the details of where each value comes from myself, am wondering if you got the chance to or not.

Reply:
With the host header, the port is optional so it may be present in some headers https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Host

I think with SNI, from various sources I checked, we should only have the hostname present (no port included) https://www.networking4all.com/en/support/faq/what-is-sni

I'm pretty confident that this should be all the gotchas, but let me add a log message if this error is hit to log the SNI and host header. This is also why I made it configurable so we can turn off the check on the server.